src/electionguard_gui/main_app.py

```diff
@@ -101,8 +101,19 @@ def start(self) -> None:
             port = self.config_service.get_port()
             host = self.config_service.get_host()
             self.log_service.debug(f"Starting eel port={port} mode={mode} host={host}")
-            eel.start("index.html", size=(1024, 768), port=port, mode=mode, host=host)
+            eel.start(
+                "index.html",
+                size=(1024, 768),
+                port=port,
+                mode=mode,
+                host=host,
+                close_callback=self.on_close,
```

Comment:
simply overriding the `close_callback` method doesn't seem like it should stop the eel from exiting on it's own, but whether it's a bug or not that's how it actually works